cmd/client-fs_linux.go

```diff
@@ -60,3 +61,20 @@ func IsDeleteEvent(event notify.Event) bool {
 	}
 	return false
 }
+
+// GetXattr returns the extended attributes for a file if supported
+// by the OS
+func GetXattr(path string) map[string][]string {
+	xMetadata := map[string][]string{}
+	if xattr.Supported(path) {
```

Comment:
We shouldn't ignore the errors in this one, other than syscall.ENOTSUP, something like this would work. 

```go
func isSysErrNotSupported(e error) bool {
	xerr, ok := e.(*xattr.Error)
	if ok {
		return xerr.Err == syscall.ENOTSUP
	}
	return false
}

func getXAttr(path, key string) (string, error) {
	data, e := xattr.Get(path, key)
	if e != nil && !isSysErrNotSupported(e) {
		return "", e
	}
	return string(data), nil
}

func getAllXattrs(path string) (map[string]string, error) {
	xMetadata := make(map[string]string)
	list, e := xattr.List(path)
	if e != nil && !isSysErrNotSupported(e) {
		return nil, e

	}
	for _, key := range list {
		xMetadata[key], e = getXAttr(path, key)
		if e != nil {
			return nil, e
		}

	}
	return xMetadata, nil
}
```

Also `map[string][]string`  can be simply `map[string]string` it is harder to manage slice `[]string`, perhaps client-s3.go API can be changed to address this as well.  We should also remove the `clientContent.Metadata` to have `map[string][]string` type. getSourceStream() seems to have a weird code like following..

```
        metadata = map[string]string{}
        if fetchStat {
                st, err := sourceClnt.Stat(false)
                if err != nil {
                        return nil, nil, err.Trace(alias, urlStr)
                }
                for k, v := range st.Metadata {
                        if len(v) > 0 {
                                metadata[k] = v[0]
                        }
                }
        }
        return reader, metadata, nil
```

This type of code can be avoided.

Reply:
ok,fixed this.